### PR TITLE
Add support for provisioning over IPv6

### PR DESCRIPTION
**Pull request by derekhiggins**
Sticking to the same defaults as the metal3-dev-env support
for consistency.

### CODE REVIEW

**Comment by derekhiggins** [comment]
For IPv6 should be tested with https://github.com/openshift/installer/pull/2727

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1390/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1418/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1426/

**Comment by stbenjam** [comment]
The installer IPv6 fixes have landed. This needs a rebase when you get a chance.

I assume we also still need the UEFI/IPv6-compatible (snponly.efi) iPXE in the OpenShift ironic images

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1502/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1511/

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1514/

File: 02_configure_host.sh [hunk 1]
```diff
@@ -39,6 +39,8 @@ ANSIBLE_FORCE_COLOR=true ansible-playbook \
     -e "num_masters=$NUM_MASTERS" \
     -e "num_workers=$NUM_WORKERS" \
     -e "extradisks=$VM_EXTRADISKS" \
+    -e "libvirt_nic_model=e1000" \
```
Comment (stbenjam) [inline 02_configure_host.sh]: Why change to e1000? I think once we get snponly.efi built, I got it working with the virtio network driver.

File: utils.sh [hunk 2]
```diff
@@ -68,6 +68,22 @@ function create_cluster() {
...
 
+function ipversion(){
+    if [[ $1 =~ : ]] ; then
+        echo 6
+        exit
+    fi
+    echo 4
+}
+
+function wrapifipv6(){
```
Comment (stbenjam) [inline utils.sh]: Nit, could we make this snake case (wrap_if_ipv6) to conform to our other function names?

File: 02_configure_host.sh (hunk 1 above)
Comment (derekhiggins) [reply 02_configure_host.sh]: ok, I'll try it out today

File: utils.sh (hunk 2 above)
Comment (derekhiggins) [reply utils.sh]: yup

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1521/

File: 02_configure_host.sh [hunk 3]
```diff
@@ -153,15 +163,15 @@ if [ "${RHEL8}" = "True" ] ; then
     sudo firewall-cmd --zone=libvirt --add-port=6230-${VBMC_MAX_PORT}/udp
     sudo firewall-cmd --zone=libvirt --add-port=6230-${VBMC_MAX_PORT}/udp --permanent
 else
-    if ! sudo iptables -C INPUT -i baremetal -p udp -m udp --dport 6230:${VBMC_MAX_PORT} -j ACCEPT 2>/dev/null ; then
-        sudo iptables -I INPUT -i baremetal -p udp -m udp --dport 6230:${VBMC_MAX_PORT} -j ACCEPT
+    if ! sudo $IPTABLES -C INPUT -i baremetal -p udp -m udp --dport 6230:6235 -j ACCEPT 2>/dev/null ; then
+        sudo $IPTABLES -I INPUT -i baremetal -p udp -m udp --dport 6230:6235 -j ACCEPT
```
Comment (stbenjam) [inline 02_configure_host.sh]: Sorry just noticed this: can you put back VBMC_MAX_PORT?

Otherwise this is good to go.

File: ocp_install_env.sh [hunk 4]
```diff
@@ -108,8 +108,9 @@ controlPlane:
 platform:
   baremetal:
 $(network_configuration)
-    bootstrapOSImage: http://${MIRROR_IP}/images/${MACHINE_OS_BOOTSTRAP_IMAGE_NAME}?sha256=${MACHINE_OS_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256}
-    clusterOSImage: http://${MIRROR_IP}/images/${MACHINE_OS_IMAGE_NAME}?sha256=${MACHINE_OS_IMAGE_SHA256}
+    bootstrapOSImage: http://$(wrap_if_ipv6 $MIRROR_IP)/images/${MACHINE_OS_BOOTSTRAP_IMAGE_NAME}?sha256=${MACHINE_OS_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256}
+    clusterOSImage: http://$(wrap_if_ipv6 $MIRROR_IP)/images/${MACHINE_OS_IMAGE_NAME}?sha256=${MACHINE_OS_IMAGE_SHA256}
+    provisioningNetworkCIDR: $PROVISIONING_NETWORK
```
Comment (stbenjam) [inline ocp_install_env.sh]: This should be removed, it's handled by the network_configuration function above

File: 02_configure_host.sh (hunk 3 above)
Comment (derekhiggins) [reply 02_configure_host.sh]: yup

File: ocp_install_env.sh (hunk 4 above)
Comment (derekhiggins) [reply ocp_install_env.sh]: will do

**Comment by stbenjam** [comment]
Thanks, LGTM

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1523/

**Pull request merged**